### ISSUE TITLE
Deprecated warnings in typst 1.12

### ISSUE THREAD

**Issue by PhyX-Meow** [platformer/typst-algorithms#22]
Though it may not be soon released but it is good to know them.
```
warning: `state.display` is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:580:2
    │
580 │ ╭   _algo-in-algo-context.display(is-in-algo => {
581 │ │     _algo-assert(is-in-algo, message: message)
582 │ │   })
    │ ╰────^
    │
    = hint: use `state.get` in a `context` expression instead

warning: `state.display` is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:601:2
    │
601 │ ╭   _algo-indent-level.display(n => {
602 │ │     _algo-assert(n - 1 >= 0, message: "dedented too much")
603 │ │   })
    │ ╰────^
    │
    = hint: use `state.get` in a `context` expression instead

warning: `state.display` is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:442:6
    │
442 │ ╭       _algo-indent-level.display(indent-level => {
443 │ │         if indent-guides != none {
444 │ │           _algo-indent-guides(
445 │ │             indent-guides,
    · │
463 │ │         ))
464 │ │       })
    │ ╰────────^
    │
    = hint: use `state.get` in a `context` expression instead

warning: `locate` with callback function is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:496:6
    │
496 │   ) = { locate(loc => {
    │ ╭───────^
497 │ │   let id-str = str(counter(_algo-id-ckey).at(loc).at(0))
498 │ │   let comment-dicts = _algo-comment-dicts.final(loc)
499 │ │   let has-comments = id-str in comment-dicts
    · │
573 │ │   )
574 │ │ })}
    │ ╰──^
    │
    = hint: use a `context` expression instead

warning: `state.display` is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:632:4
    │
632 │ ╭     _algo-comment-prefix.display(comment-prefix => {
633 │ │       _algo-comment-styles.display(comment-styles => {
634 │ │         set text(..comment-styles)
635 │ │         comment-prefix
636 │ │         no-emph(body)
637 │ │       })
638 │ │     })
    │ ╰──────^
    │
    = hint: use `state.get` in a `context` expression instead

warning: calling `state.final` with a location is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:498:22
    │
498 │   let comment-dicts = _algo-comment-dicts.final(loc)
    │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    │
    = hint: try removing the location argument

warning: `locate` with callback function is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:238:6
    │
238 │   ) = { locate(loc => style(styles => {
    │ ╭───────^
239 │ │   let id-str = str(counter(_algo-id-ckey).at(loc).at(0))
240 │ │   let line-index-str = str(line-index)
241 │ │   let comment-dicts = _algo-comment-dicts.final(loc)
    · │
306 │ │   )
307 │ │ }))}
    │ ╰───^
    │
    = hint: use a `context` expression instead

warning: `style` is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:238:20
    │
238 │   ) = { locate(loc => style(styles => {
    │ ╭─────────────────────^
239 │ │   let id-str = str(counter(_algo-id-ckey).at(loc).at(0))
240 │ │   let line-index-str = str(line-index)
241 │ │   let comment-dicts = _algo-comment-dicts.final(loc)
    · │
306 │ │   )
307 │ │ }))}
    │ ╰──^
    │
    = hint: use a `context` expression instead

warning: calling `state.final` with a location is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:241:22
    │
241 │   let comment-dicts = _algo-comment-dicts.final(loc)
    │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    │
    = hint: try removing the location argument

warning: calling `measure` with a styles argument is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:248:4
    │
248 │ ╭     measure(
249 │ │       {
250 │ │         set text(..main-text-styles)
251 │ │         _alphanumerics
    · │
254 │ │       styles
255 │ │     ).height,
    │ ╰─────^
    │
    = hint: try removing the styles argument

warning: calling `measure` with a styles argument is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:258:4
    │
258 │ ╭     measure(
259 │ │       {
260 │ │         set text(..comment-styles)
261 │ │         comment-content
262 │ │       },
263 │ │       styles
264 │ │     ).height,
    │ ╰─────^
    │
    = hint: try removing the styles argument

warning: calling `measure` with a styles argument is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:267:4
    │
267 │ ╭     measure(
268 │ │       {
269 │ │         set text(..line-number-styles)
270 │ │         _numerals
271 │ │       },
272 │ │       styles
273 │ │     ).height
    │ ╰─────^
    │
    = hint: try removing the styles argument

warning: calling `measure` with a styles argument is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:277:24
    │
277 │     let indent-size-abs = measure(
    │ ╭─────────────────────────^
278 │ │     rect(width: indent-size),
279 │ │     styles
280 │ │   ).width
    │ ╰───^
    │
    = hint: try removing the styles argument

warning: calling `measure` with a styles argument is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:282:24
    │
282 │     let block-inset-abs = measure(
    │ ╭─────────────────────────^
283 │ │     rect(width: block-inset),
284 │ │     styles
285 │ │   ).width
    │ ╰───^
    │
    = hint: try removing the styles argument

warning: calling `measure` with a styles argument is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:287:23
    │
287 │     let row-gutter-abs = measure(
    │ ╭────────────────────────^
288 │ │     rect(width: row-gutter),
289 │ │     styles
290 │ │   ).width
    │ ╰───^
    │
    = hint: try removing the styles argument

warning: `state.display` is deprecated
    ┌─ @preview/algo:0.3.3/algo.typ:633:6
    │
633 │ ╭       _algo-comment-styles.display(comment-styles => {
634 │ │         set text(..comment-styles)
635 │ │         comment-prefix
636 │ │         no-emph(body)
637 │ │       })
    │ ╰────────^
    │
    = hint: use `state.get` in a `context` expression instead
```

**Comment by platformer** [comment]
Thanks for letting me know. I was still hoping for custom types to land before I really made a big update to the package, but I'll go ahead and update the package before the next major Typst version arrives

**Comment by NonDairyNeutrino** [comment]
I would also love to see a fix for this.  Maybe I'm missing a compiler flag or something, but due to this issue, `#algo` is broken and unusable.  Is there a known workaround at the moment?  Or is the fix straightforward enough to warrant submitting a pull request?

**Comment by platformer** [comment]
> I would also love to see a fix for this. Maybe I'm missing a compiler flag or something, but due to this issue, `#algo` is broken and unusable.

Do you mean it's outright nonfunctional? My test cases look about the same as they always have, but I want to make sure there's nothing else that needs to be done besides migrating the package to the new context syntax

**Comment by NonDairyNeutrino** [comment]
Hmm, actually upon further inspection, it might be not related to this issue after all.  I'll make a new issue though.

**Comment by platformer** [comment]
fixed by 7b89e81
